modules/tinymce/src/core/test/ts/atomic/html/Base64UrisTest.ts

```diff
@@ -98,6 +98,72 @@ describe('atomic.tinymce.core.html.Base64UrisTest', () => {
         }
       }
     );
+
+    testExtract(
+      'TINY-8646: Should extract base64 encoded image following `data:` text on the same paragaph',
+      '<p>my data:<img src="data:image/gif;base64,R0/yw=="/></p>',
+      {
+        html: '<p>my data:<img src="$prefix_0"/></p>',
+        uris: {
+          $prefix_0: 'data:image/gif;base64,R0/yw=='
+        }
+      }
+    );
+
+    testExtract(
+      'TINY-8646: Should extract base64 encoded image in the next paragraph following `data:` text',
+      '<p>my data:</p><p><img src="data:image/gif;base64,R0/yw=="/></p>',
+      {
+        html: '<p>my data:</p><p><img src="$prefix_0"/></p>',
+        uris: {
+          $prefix_0: 'data:image/gif;base64,R0/yw=='
+        }
+      }
+    );
+
+    testExtract(
+      'TINY-8646: Should extract based64 encoded image following an attribute with `data:` text as a value',
+      '<img name="data:" src="data:image/gif;base64,R0/yw=="/>',
+      {
+        html: '<img name="data:" src="$prefix_0"/>',
+        uris: {
+          $prefix_0: 'data:image/gif;base64,R0/yw=='
+        }
+      }
+    );
+
+    testExtract(
+      'TINY-8646: Should extract based64 encoded image when there is no space between attributes',
+      '<img name="data:"src="data:image/gif;base64,R0/yw=="/>',
+      {
+        html: '<img name="data:"src="$prefix_0"/>',
+        uris: {
+          $prefix_0: 'data:image/gif;base64,R0/yw=='
+        }
+      }
+    );
+
+    testExtract(
+      'TINY-8646: Should extract based64 encoded image when single quotes are used',
+      '<img name=\'data:\'src=\'data:image/gif;base64,R0/yw==\'/>',
```

Comment:
It might be more readable if you use template literals here instead?
```suggestion
      `<img name='data:'src='data:image/gif;base64,R0/yw=='/>`,
```